custom_components/e3dc_rscp/sensor.py

```diff
@@ -372,6 +372,13 @@
         device_class=SensorDeviceClass.POWER_FACTOR,
         state_class=SensorStateClass.MEASUREMENT,
     ),
+    SensorEntityDescription(
+        key="smartgrid",
+        translation_key="smartgrid",
+        icon="mdi:solar-power",
+        device_class=SensorDeviceClass.ENUM,
+        options=["blocked", "normal", "recommendation_on", "startup"],
```

Comment:
@tobiby thanks for this addition, I think that is quite valuable. So far I've only one question: Will these strings be translated by HA? Otherwise this should be adapted accordingly, I'd have to look that up myself (don't have the time for it yet), but that should be in it, so that we can later add a translated version of it as well.

Reply:
I got the names of the states from the smartgrid specifications.
If a translation of the states is desired, you would then have to find the corresponding translations. is that useful / necessary?

Reply:
Translation should be straight forward, see example.
@tobiby Can you test this translation (to be included in `strings.json` and `translations/en.json`) and if this works update your branch?

```JSON
    "state": {
        "sensor": {
            "my_enum_sensor": {
                "blocked": "blocked mode",
                "normal": "normal operation",
                "recommendation_on": "PV surplus operation"
                "startup": "Operation for derating"
            }
        }
    }

Reply:
Added the translation to the named files. I currently dont see any issues with it. But maybe someone else can do a quick test, too. Thanks!